File: Benchmark/wrk/kv/insert.lua
-- Warning: This script is designed for single-threaded use only.

counter = 0

-- FNV-1a 32-bit hash
function fnv1a_hash(x)
  local hash = 2166136261
  local prime = 16777619
  local s = tostring(x)
  for i = 1, #s do
    hash = hash ~ string.byte(s, i)
    hash = (hash * prime) % 2^32
  end
  return tostring(hash)
end

function request()
  counter = counter + 1
  local key = fnv1a_hash(counter)
  local value = "val" .. key

  return wrk.format("POST", "/kv/" .. key,
    {["Content-Type"]="text/plain"},
    value)
end

File: Benchmark/k6/kv/insert.js
import http from "k6/http";

let counter = 0;

function fnv1aHash(x) {
  let hash = 2166136261;
  const prime = 16777619;
  const s = x.toString();
  for (let i = 0; i < s.length; i++) {
    hash ^= s.charCodeAt(i);
    hash = (hash * prime) >>> 0;
  }
  return hash.toString();
}

export let options = { vus: 50, duration: "120s" };

const base = __ENV.K6_BASE_URL || __ENV.BASE_URL || "http://127.0.0.1:8080";

export default function () {
  counter++;
  let key = fnv1aHash(counter);
  let value = "val" + key;
  http.post(`${base}/kv/${key}`, value, {
    headers: { "Content-Type": "text/plain" },
  });
}
